io/net/virt-net/network_virtualization.py

```diff
@@ -433,30 +433,40 @@ def test_vnic_dlpar(self):
             dev_id = self.find_device_id(mac)
             device_name = self.find_device(mac)
             slot = self.find_virtual_slot(dev_id)
+            networkinterface = NetworkInterface(device_name, self.local)
             if slot:
                 try:
                     for _ in range(self.num_of_dlpar):
                         self.drmgr_vnic_dlpar('-r', slot)
+                        if self.interface_present(device_name):
+                            self.fail("DLPAR remove did not remove interface")
                         self.drmgr_vnic_dlpar('-a', slot)
-                        self.wait_intrerface(device_name)
+                        self.wait_interface(device_name)
+                        try:
+                            networkinterface.add_ipaddr(device_ip, netmask)
+                            networkinterface.bring_up()
```

Comment:
Please note that I added the command to bring up the interface up. When the interface is added back, it is in a DOWN state and the function `networkinterface.is_link_up` does not bring the interface up, rather checks if the interface is up.

Reply:
@cforno12 looks fine, could you also add a line to log the iteration count, it will be easy to know if dlpar fails for which iteration step?

Reply:
@cforno12 suggest to add networkinterface.bring_up() after try...except block so it will bring up interface after ip added.